generated/language/src/test/java/com/google/cloud/language/v1/spring/LanguageAutoConfigurationTests.java

```diff
@@ -0,0 +1,310 @@
+  void testQuotaProjectIdFromProperties() {
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.cloud.language.v1.spring.auto.language-service.quota-project-id="
+                + SERVICE_OVERRIDE_CLIENT_ID)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              String quotaProjectId = client.getSettings().getQuotaProjectId();
+              assertThat(quotaProjectId).isEqualTo(SERVICE_OVERRIDE_CLIENT_ID);
+            });
+  }
+
+  @Test
+  void testExecutorThreadCountFromProperties() {
+    this.contextRunner
+        .withPropertyValues(
+            "com.google.cloud.language.v1.spring.auto.language-service.executor-thread-count=3")
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              InstantiatingExecutorProvider backgroundExecutorProvider =
+                  ((InstantiatingExecutorProvider)
+                      client.getSettings().getBackgroundExecutorProvider());
+              assertThat(backgroundExecutorProvider.toBuilder().getExecutorThreadCount())
+                  .isEqualTo(3);
+            });
+  }
+
+  @Test
+  void testCustomTransportChannelProviderUsedWhenProvided() throws IOException {
+    when(mockTransportChannelProvider.getTransportName()).thenReturn("grpc");
+    when(mockTransportChannelProvider.getTransportChannel()).thenReturn(mockTransportChannel);
+    when(mockTransportChannel.getEmptyCallContext()).thenReturn(mockApiCallContext);
+    when(mockApiCallContext.withCredentials(any())).thenReturn(mockApiCallContext);
+    when(mockApiCallContext.withTransportChannel(any())).thenReturn(mockApiCallContext);
+
+    contextRunner
+        .withBean(
+            "defaultLanguageServiceTransportChannelProvider",
+            TransportChannelProvider.class,
+            () -> mockTransportChannelProvider)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              assertThat(client.getSettings().getTransportChannelProvider())
+                  .isSameAs(mockTransportChannelProvider);
+            });
+  }
+
+  @Test
+  void testCustomServiceSettingsUsedWhenProvided() throws IOException {
+    String mockQuotaProjectId = "mockQuotaProjectId";
+    LanguageServiceSettings customLanguageServiceSettings =
+        LanguageServiceSettings.newBuilder()
+            .setCredentialsProvider(mockCredentialsProvider)
+            .setQuotaProjectId(mockQuotaProjectId)
+            .build();
+    contextRunner
+        .withBean(
+            "languageServiceSettings",
+            LanguageServiceSettings.class,
+            () -> customLanguageServiceSettings)
+        .run(
+            ctx -> {
+              LanguageServiceClient client = ctx.getBean(LanguageServiceClient.class);
+              assertThat(client.getSettings().getCredentialsProvider())
```

Comment:
Can you just assert that `client.getSettings()` is the same as `customLanguageServiceSettings`?

Reply:
I somehow missed this, thanks for catching! Updated.